### ISSUE TITLE
Add SerializeAsAny to migration guide and possibly move functionality into model_config 

### ISSUE THREAD

**Issue by taddes** [pydantic/pydantic#7043]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

We ran into some issues with serialization when migrating (subclass data loss). It wasn't until finding [this bug ](https://github.com/pydantic/pydantic/issues/6895) that led us to discover the docs with the [Migration warning ](https://docs.pydantic.dev/latest/usage/serialization/#serializing-subclasses) stating you need to use SerializeAsAny. 

This is less a bug and more a request to put this front and center in the migration doc. We lost a lot of time trying to figure this out and given that serialization is so essential for any API, esp FastAPI, this isn't something that should require a ton of searching to figure out.

Anyway, wondering if there is a possible plan to implement this logic within the `model_config = ConfigDict()`? I think it would make more intuitive sense.

From Docs, perhaps add to migration guide:
```This behavior is different from how things worked in Pydantic V1, where we would always include all (subclass) fields when recursively dumping models to dicts. The motivation behind this change in behavior is that it helps ensure that you know precisely which fields could be included when serializing, even if subclasses get passed when instantiating the object. In particular, this can help prevent surprises when adding sensitive information like secrets as fields of subclasses.```

Thanks so much!

### Python, Pydantic & OS Version

```Text
pydantic version: 2.1.0
        pydantic-core version: 2.4.0
          pydantic-core build: profile=release pgo=false mimalloc=true
                 install path: /Users/taddeskorris/.pyenv/versions/3.11.0/envs/merino-311/lib/python3.11/site-packages/pydantic
               python version: 3.11.0 (main, Dec 13 2022, 15:17:55) [Clang 13.1.6 (clang-1316.0.21.2.5)]
                     platform: macOS-12.5-arm64-arm-64bit
     optional deps. installed: ['typing-extensions']
```

Selected Assignee: @dmontagu

**Comment by adriangb** [comment]
> Anyway, wondering if there is a possible plan to implement this logic within the model_config = ConfigDict()? I think it would make more intuitive sense.

We decided it would be better as a runtime flag, see https://github.com/pydantic/pydantic-core/pull/740. The constraint has been bandwidth in implementation. A contribution to move that work forward or to improve the docs would be very welcome!

**Comment by sydney-runkle** [comment]
Closing as a duplicate of https://github.com/pydantic/pydantic/issues/6423, which we've included in our `2.6.0` milestone. Will remember to update the migration guide and docs as well 👍
